c3.php

```diff
@@ -8,420 +8,463 @@
+                } else {
+                    $settings = Configuration::config();
+                }
 
-        return [$phpCoverage, $file];
-    }
+                $pathCoverage = false;
+                if (isset($settings['coverage']['path_coverage'])) {
+                    $pathCoverage = (bool)$settings['coverage']['path_coverage'];
+                }
 
-    function __c3_exit()
-    {
-        if (!isset($_SERVER['HTTP_X_CODECEPTION_CODECOVERAGE_DEBUG'])) {
-            exit;
-        }
-        return null;
-    }
+                if (class_exists(Selector::class)) {
+                    //php-code-coverage >= 9.1.10
+                    $filter = new CodeCoverageFilter();
+                    if ($pathCoverage) {
+                        $driver = (new Selector())->forLineAndPathCoverage($filter);
+                    } else {
+                        $driver = (new Selector())->forLineCoverage($filter);
+                    }
+                    $phpCoverage = new CodeCoverage($driver, $filter);
+                } elseif (method_exists(Driver::class, 'forLineCoverage')) {
+                    //php-code-coverage 9.0.0 - 9.1.9
+                    $filter = new CodeCoverageFilter();
+                    if ($pathCoverage) {
+                        $driver = Driver::forLineAndPathCoverage($filter);
+                    } else {
+                        $driver = Driver::forLineCoverage($filter);
+                    }
+                    $phpCoverage = new CodeCoverage($driver, $filter);
+                } else {
+                    //php-code-coverage 8 or older
+                    $phpCoverage = new PHP_CodeCoverage();
+                }
 
-    function __c3_clear()
-    {
-        \Codeception\Util\FileSystem::doEmptyDir(C3_CODECOVERAGE_MEDIATE_STORAGE);
-    }
-}
+                try {
+                    \Codeception\Coverage\Filter::setup($phpCoverage)
+                        ->whiteList($settings)
+                        ->blackList($settings);
+                } catch (Exception $e) {
+                    __c3_error($e->getMessage());
+                }
 
-if (!is_dir(C3_CODECOVERAGE_MEDIATE_STORAGE)) {
-    if (mkdir(C3_CODECOVERAGE_MEDIATE_STORAGE, 0777, true) === false) {
-        __c3_error('Failed to create directory "' . C3_CODECOVERAGE_MEDIATE_STORAGE . '"');
-    }
-}
+                return [$phpCoverage, $file];
+            }
 
-// evaluate base path for c3-related files
-$path = realpath(C3_CODECOVERAGE_MEDIATE_STORAGE) . DIRECTORY_SEPARATOR . 'codecoverage';
+            function __c3_exit()
+            {
+                if (!isset($_SERVER['HTTP_X_CODECEPTION_CODECOVERAGE_DEBUG'])) {
+                    exit;
+                }
+                return null;
+            }
 
-$requestedC3Report = (strpos($_SERVER['REQUEST_URI'], 'c3/report') !== false);
+            function __c3_clear()
+            {
+                \Codeception\Util\FileSystem::doEmptyDir(C3_CODECOVERAGE_MEDIATE_STORAGE);
```

Comment:
We should not be using this constant.
Instead:
- define it as a constructor argument 
- store it in a private property
- in the constructor check if the constant is defined and if so use it to initialize the private property
- remove all references to this constant